src/EditorFeatures/CSharpTest2/Recommendations/WhenKeywordRecommenderTests.cs

```diff
@@ -63,5 +63,13 @@ public class WhenKeywordRecommenderTests : KeywordRecommenderTests
             await VerifyAbsenceAsync(AddInsideMethod(
 @"try {} catch (Exception e) when (true) $$"));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
```

Comment:
:bulb: Consider adding tests corresponding to each of the edge cases added in `IsValidContext`. Ideally one or more of the tests should fail if any of the `if` statements is removed.

Reply:
This test already covers all those cases (it includes multiple variants, typing the keyword one character at a time) :-)